src/components/scene/ar-hit-test.js

```diff
@@ -1,5 +1,29 @@
+var register = require('../../core/component').registerComponent;
+var THREE = require('../../lib/three');
+var CAM_LAYER = 21;
+var previousFrameAnchors = new Set();
+var anchorToObject3D = new Map();
+var hitTestCache;
+var tempVec3 = new THREE.Vector3();
+var tempQuaternion = new THREE.Quaternion();
+var fakePose = {
+  transform: {
+    orientation: new THREE.Quaternion(),
+    position: new THREE.Vector3()
+  }
+};
+
+function applyPose (pose, object3D, offset) {
+  object3D.position.copy(pose.transform.position);
+  object3D.quaternion.copy(pose.transform.orientation);
+
+  tempVec3.copy(offset);
+  tempQuaternion.copy(pose.transform.orientation);
+  tempVec3.applyQuaternion(tempQuaternion);
+  object3D.position.sub(tempVec3);
```

Comment:
Is `sub` more clear than `add` in this context? I usually associate `offset` with `addition` but don't fully understand yet the full context here.

Reply:
Ah i'm not adding an additional offset.

At the beginning I calculate the offset the object's position versus the center of the calculated reticle.
I subtract this offset when the user places the object so that it lines up optically with the reticle